tests/recipes/cpc2/test_compute_haspi.py

```diff
@@ -0,0 +1,42 @@
+"""Tests for the haspi computation functions."""
+
+import numpy as np
+import pytest
+
+from recipes.cpc2.baseline.compute_haspi import (
+    parse_cec2_signal_name,
+    set_seed_with_string,
+)
+
+
+@pytest.mark.parametrize(
+    "signal_name, expected",
+    [("S1_L1_E1_hr", ("S1", "L1", "E1_hr")), ("S1_L1_E2", ("S1", "L1", "E2"))],
+)
+def test_parse_cec2_signal_name_ok(signal_name, expected):
+    """Test the parse_CEC2_signal_name function."""
+    assert parse_cec2_signal_name(signal_name) == expected
+
+
+@pytest.mark.parametrize(
+    "signal_name, expected",
+    [
+        ("S1", ValueError),
+        ("S1_L1", ValueError),
+        ("___", ValueError),
+        ("_X_X", ValueError),
+    ],
+)
+def test_parse_cec2_signal_name_error(signal_name, expected):
+    """Test the parse_CEC2_signal_name function for invalid inputs."""
+    with pytest.raises(expected):
+        parse_cec2_signal_name(signal_name)
+
+
+@pytest.mark.parametrize("string_value", ["", "abc", "123", "abc123"])
+def test_set_seed_with_string_ok(string_value):
+    """Test the set_seed_with_string function."""
+    set_seed_with_string(string_value)
+    x = np.random.randint(0, 100)
+    set_seed_with_string(string_value)
```

Comment:
Not clear to me, but I'd guess this is needed to reset the seed so that the following `np.random.randint(0, 100)` is done with the same seed anew?

Reply:
Yeah, I think is only testing that setting the seed to a specific value is working as expected. so both np.random.randint(0, 100 ) should return the same value.
This to always use the same seed when evaluating a single scene

Reply:
Yes, it's just testing that the set_seed_with_string() function runs and that it causes np.random to return the same value. Will change the 100 to something higher because there currently 1% chance the test would pass even with different seend 👍